java/arcs/core/data/RawEntity.kt

```diff
@@ -24,11 +24,14 @@ data class RawEntity(
      * Collection ([Set]) fields and the set of [ReferenceId]s referencing the values in those
      * collections.
      */
-    val collections: Map<FieldName, Set<Referencable>> = emptyMap()
+    val collections: Map<FieldName, Set<Referencable>> = emptyMap(),
+    /** Indication of the timestamp when this entity expires. */
+    val expirationTimestamp: String = ""
```

Comment:
We will need an interface for a cross-platform timestamp generator that the Database and RamDisk can use to compare existing timestamps against. Ideally this would be something lexicographically sortable, or even better: a number - so we don't have to implement custom logic outside of the queries to the database when determining what to delete.

Reply:
won't a regular `Date` class work in both cases?

Reply:
Yeah that would be fine. A string is probably too ambiguous though.

Reply:
There doesn't seem to be a Cross-platform Date class in the kotlin stdlib.